internal/setup/setup.go

```diff
@@ -699,3 +738,75 @@ func Select(release *Release, slices []SliceKey) (*Selection, error) {
 
 	return selection, nil
 }
+
+// pathInfoToYAML converts a PathInfo object to a yamlPath object.
+func pathInfoToYAML(pi *PathInfo) (*yamlPath, error) {
+	if pi == nil {
+		return nil, nil
+	}
+	path := &yamlPath{
+		Mode:    yamlMode(pi.Mode),
+		Mutable: pi.Mutable,
+		Until:   pi.Until,
+		Arch:    yamlArch{List: pi.Arch},
+	}
+	switch pi.Kind {
+	case DirPath:
+		path.Dir = true
+	case CopyPath:
+		path.Copy = pi.Info
+	case TextPath:
+		path.Text = &pi.Info
+	case SymlinkPath:
+		path.Symlink = pi.Info
+	case GlobPath:
+	default:
+		return nil, fmt.Errorf("internal error: unrecognised PathInfo type: %s", pi.Kind)
+	}
+	return path, nil
+}
+
+// sliceToYAML converts a Slice object to a yamlSlice object.
+func sliceToYAML(s *Slice) (*yamlSlice, error) {
+	if s == nil {
+		return nil, nil
+	}
+	slice := &yamlSlice{
+		Essential: make([]string, len(s.Essential)),
```

Comment:
I find it is simpler to do:
```go
Essential: make([]string, 0, len(s.Essential))
```
and then use append normally in the code below instead of using `i`.